src/NUnitFramework/framework/Internal/Commands/BeforeAndAfterTestCommand.cs

```diff
@@ -52,11 +52,25 @@ public override TestResult Execute(TestExecutionContext context)
             if (Test.Fixture == null)
                 Test.Fixture = context.TestObject;
 
-            try
+            ThreadAbortSafeRun(context, () =>
             {
                 BeforeTest(context);
-
                 context.CurrentResult = innerCommand.Execute(context);
+            });
+
+            if (context.ExecutionStatus != TestExecutionStatus.AbortRequested)
+            {
+                ThreadAbortSafeRun(context, () => { AfterTest(context); });
+            }
+
+            return context.CurrentResult;
+        }
+
+        private static void ThreadAbortSafeRun(TestExecutionContext context, Action action)
```

Comment:
Can we find a name for this that makes it clear it is only intended for running test methods under the execution context? I realize it's private to the class, but the name makes it sound more general than it really is, since it records test results in the context.

Since I don't actually have a name to suggest, you can take this as a suggestion and leave as is if you can't think of a better name. :-)

I think this may resolve #352 as well.

Reply:
What about one of these?
- `RunWithContextInThreadAbortSafeZone`
- `TestMethodRunInThreadAbortSafeZone`
- `TestMethodContextRunInThreadAbortSafeZone`
- `TestContextInThreadAbortSafeZone`